base_layer/core/src/transactions/helpers.rs

```diff
@@ -383,7 +384,7 @@ pub fn schema_to_transaction(txns: &[TransactionSchema]) -> (Vec<Arc<Transaction
 /// ```
 pub fn display_currency(value: f64, precision: usize, separator: &str) -> String {
     let whole = value as usize;
-    let decimal = ((value - whole as f64) * 10_f64.powf(precision as f64)).round() as usize;
+    let decimal = ((value - whole as f64) * pow(10_f64, precision)).round() as usize;
```

Comment:
```suggestion
    let decimal = ((value - whole as f64) * pow(10, precision)).round() as usize;
```

Reply:
It wants a safe cast to f64, thus either `10_f64` or `10 as f64`